doc/source/development/rfc/rfc94_field_precision_width_metadata.rst

```diff
@@ -0,0 +1,136 @@
+.. _rfc-94:
+
+=============================================================
+RFC 94: Numeric fields width/precision metadata
+=============================================================
+
+============== =============================================
+Author:        Alessandro Pasotti
+Contact:       elpaso @ itopen.it
+Started:       2023-Feb-17
+Status:        Development
+Target:        GDAL 3.7
+============== =============================================
+
+Summary
+-------
+
+The document proposes and describes the introduction of a new method
+to the `OGRFieldDefn` class that returns information for numeric real
+fields about how the precision and width have been calculated by the
+driver and have to be interpreted.
+
+Motivation
+----------
+
+Applications (for example OGR/GDAL utils and QGIS) may require information
+about the field width and precision in order to convert between different
+formats or to validate user input on editing.
+
+Different drivers may calculate the width and precision differently by including
+or not the decimal separator and/or the minus sign into the calculated length.
+
```

Comment:
We should try to establish in this RFC a table with the situation for drivers that use SetPrecision(), like Shapefile, MapInfo, PostgreSQL (are all SQL drivers the same like MySQL, MSSQL, OCI) , GPKG (this one will be N/A), CSV (from the .csvt side car), HANA (@mrylov can hopefully help us clarify the semantics), FlatGeoBuf, MEM, FileGDB, GML to see if we are not missing some subtelties

Reply:
Good idea! I have added the table placing a `?` where I do not know or I am not sure.